storage/aptosdb/src/pruner/mod.rs

```diff
@@ -135,27 +142,39 @@ impl Pruner {
     }
     /// Sends pruning command to the worker thread when necessary.
     pub fn maybe_wake_pruner(&self, latest_version: Version) {
+        if self.last_version_sent_to_pruners.is_none() {
+            return;
+        }
         *self.latest_version.lock() = latest_version;
         if latest_version
-            >= *self.last_version_sent_to_pruners.lock() + self.pruning_batch_size as u64
+            >= *self.last_version_sent_to_pruners.as_ref().unwrap().lock()
+                + self.pruning_batch_size as u64
         {
             self.wake_pruner(latest_version);
-            *self.last_version_sent_to_pruners.lock() = latest_version;
+            *self.last_version_sent_to_pruners.as_ref().unwrap().lock() = latest_version;
         }
     }
 
     fn wake_pruner(&self, latest_version: Version) {
-        let min_readable_state_store_version =
-            latest_version.saturating_sub(self.state_store_prune_window);
-        let min_readable_ledger_version = latest_version.saturating_sub(self.ledger_prune_window);
+        let mut prune_window_vector = Vec::new();
+        if let Some(state_store_prune_window) = self.state_store_prune_window {
```

Comment:
I guess @lightmark might have a better way to avoid duplicated "prune_window_vector.push"?

Reply:
Done